app/dawn-hub/Views/OnboardingViews/Step0_WelcomeView/WelcomeView.jsx

```diff
@@ -24,12 +24,32 @@ import { WELCOME } from '../../OnboardingView/OnboardingConstants';
 const WelcomeView = (props) => {
 	const { actions } = props;
 	const { setSetupStep } = actions;
+
+	const [getUserResolved, setGetUserResolved] = useState(false);
+
+	const gateSetupStep = (e) => {
+		if (getUserResolved) {
+			setSetupStep({ setup_step: WELCOME, origin: WELCOME });
+		} else {
+			e.preventDefault();
+		}
+	};
+
+	useEffect(() => {
+		actions.getUser();
+		const timer = setTimeout(() => setGetUserResolved(true), 500);
+
+		return () => {
+			clearTimeout(timer);
+		};
+	}, []);
+
```

Comment:
Yea, I agree eventually having some UI/UX feedback to the user for this would be ideal on the off chance that it happens. I've raised the timeout delay for now though 👍